app/src/main/kotlin/com/wire/android/migration/MigrationManager.kt

```diff
@@ -116,7 +132,10 @@ class MigrationManager @Inject constructor(
                     }
                 }
             }.fold({
-                MigrationData.Result.Failure
+                when (it) {
+                    is NetworkFailure -> MigrationData.Result.Failure.NoNetwork
```

Comment:
I wasn't sure as well because sometimes it may be a temporary issue that only requires retrying but you're right and it's more safe to make user log in again than to become stuck in an endless loop and actually make the app unusable.
I changed it to only pass `NetworkFailure.NoNetworkConnection` to be the `NoNetwork` failure that can be retried.